### PR TITLE
EDM-546: Fix device spec comparisons

### PR DESCRIPTION
**Pull request by avishayt**
There are a couple instances where we need to understand if device specs are being updated by the user.  reflect.DeepEqual doesn't work properly here because oneOf is translated to json.RawMessage, and comparing JSONs isn't reliable because properties aren't always sorted the same way. Instead, we use reflect.DeepEqual where possible, and unmarshal when necessary.

### CODE REVIEW

File: internal/store/common.go [hunk 1]
```diff
@@ -297,3 +299,158 @@ func retryUpdate(fn func() (bool, error)) error {
...
+
+func configsAreEqual(c1, c2 *[]api.DeviceSpec_Config_Item) bool {
+	if c1 == nil {
+		return c2 == nil
+	}
+	if c2 == nil {
+		return false // c2 is nil, c1 is not
+	}
+
+	if len(*c1) != len(*c2) {
```
Comment (ori-amizur) [inline internal/store/common.go:311]: The first 2 conditions can be skipped and use only this one

File: internal/store/common.go [hunk 2]
```diff
@@ -297,3 +299,158 @@ func retryUpdate(fn func() (bool, error)) error {
 	}
 	return err
 }
+
+func configsAreEqual(c1, c2 *[]api.DeviceSpec_Config_Item) bool {
```
Comment (ori-amizur) [inline internal/store/common.go:303]: Maybe we can use this generic func https://pkg.go.dev/slices#EqualFunc as an envelope for all the functions?

Comment (avishayt) [reply internal/store/common.go:305]: Nice.  Done.

File: internal/store/common.go (hunk 1 above)
Comment (avishayt) [reply internal/store/common.go]: Fixed.

**Review by ori-amizur — APPROVED** [review]

**Pull request merged**